### PR TITLE
[ci] #2732: Add a conditions to update iroha2-base images and add PR labels

### LINKED ISSUE #2732: Add a condition to update `iroha2-base` images
### Feature request

Don't build and publish the base image on every merge

### Motivation

CI optimization

### Who can help?

_No response_

### PR DESCRIPTION
**Pull request by BAStos525**
### Description of the Change
1. Modify iroha2 CI in order to enable conditional step of `iroha2-base` image build if its `Dockerfile.base` has been changed.
2. Get back `I2::Dev::Label` workflow since we are able to apply the same approach as for `base-image` build.
3. Remove necessaries iroha2-dev-pr-label-config.yml` and `iroha2-dev-pr-label-api.yml` workflows files.

### Issue
Since `tj-actions/changed-files` and `dorny/paths-filter` Actions doesn't work within `hyperledger/iroha` repository, this is another one possible solution to resolve #2732.

### Benefits
1. Refactoring of iroha2 CI.
2. Reduce the amount of workflows files.

### Possible Drawbacks

None

### CODE REVIEW

File: .github/workflows/iroha2-dev-pr-label.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: I2::Dev::Label
```
Comment (appetrosyan) [inline .github/workflows/iroha2-dev-pr-label.yml:1]: This is much clearer now.

**Comment by appetrosyan** [comment]
In principle I'll approve, but we should have a positive and negative test to demonstrate that it actually works when we have those changes in the CI. @6r1d what do you think? How can we arrange for something like that.

**Comment by BAStos525** [comment]
> In principle I'll approve, but we should have a positive and negative test to demonstrate that it actually works when we have those changes in the CI. @6r1d what do you think? How can we arrange for something like that.

 I usually try to test my PR on fork whenever it's possible.

image

image

image

image

**Review by appetrosyan — DISMISSED** [review]

File: .github/workflows/iroha2-release.yml [hunk 2]
```diff
@@ -31,8 +31,11 @@ jobs:
           registry: docker.soramitsu.co.jp
           username: ${{ secrets.HARBOR_USERNAME }}
           password: ${{ secrets.HARBOR_TOKEN }}
+      - name: Check changed files
+        uses: jitterbit/get-changed-files@v1
+        id: base_image
       - name: Build and push iroha2-base image
-        # TODO Add `if` condition #2732
+        if: contains(steps.base_image.outputs.added_modified, 'Dockerfile.base')
```
Comment (s8sato) [inline .github/workflows/iroha2-release.yml:38]: #2732 can be closed, but in #2690 please consider the layer caching instead of having the base image

**Review by s8sato — APPROVED** [review]

**Review by appetrosyan — APPROVED** [review]

File: .github/workflows/iroha2-release.yml (hunk 2 above)
Comment (s8sato) [reply .github/workflows/iroha2-release.yml:38]: [Failing test](https://github.com/hyperledger/iroha/actions/runs/3099709697/jobs/5019371879)

Comment (BAStos525) [reply .github/workflows/iroha2-release.yml:38]: I think this failed test is not related to the current PR changes.

**Review by s8sato — APPROVED** [review]

**Pull request merged**